sql/item_subselect.cc

```diff
@@ -3298,6 +3299,14 @@ bool Item_exists_subselect::exists2in_processor(void *opt_arg)
     set possible optimization strategies
   */
   in_subs->emb_on_expr_nest= emb_on_expr_nest;
+
+  {
+  OPT_TRACE_TRANSFORM(thd, trace_wrapper, trace_transform,
+                     in_subs->get_select_lex()->select_number,
+                      "EXISTS (SELECT)", "IN (SELECT)");
+  trace_transform.add( "upper_not", ( upper_not?true:false ) );
```

Comment:
Wouldn't just `upper_not` as second arg be sufficient.

Reply:
`Item_func_not *upper_not;`

so, not really.